### PR TITLE
Added support for C++17

### PR DESCRIPTION
**Pull request by bryancall**

### CODE REVIEW

File: lib/ts/string_view.h [hunk 1]
```diff
@@ -35,12 +35,18 @@
...
+#if __cplusplus < 201402L
 #define CONSTEXPR14 inline
 #else
 #define CONSTEXPR14 constexpr
 #endif
 
+#if __cplusplus >= 201703L
+namespace ts
+{
+typedef std::string_view string_view;
```
Comment (SolidWallOfCode) [inline lib/ts/string_view.h]: `using string_view = std::string_view;`

**Comment by SolidWallOfCode** [comment]
If you're going to much with this, I think you should add a tweak to handle the `-Wno-except-type` issue, based on the compiler version.

**Comment by d2r** [comment]
[approve ci autest]

**Comment by SolidWallOfCode** [comment]
Looks like an actual issue with autest, not a transient.
```
REGRESSION TEST SDK_API_OVERRIDABLE_CONFIGS started
[Mar 26 21:36:07.203] {0x2ac3a052d880} NOTE: crashlog started, target=31913, debug=false syslog=true, uid=989 euid=989
[Mar 26 21:36:07.203] {0x2ac3a052d880} WARNING: failed to intialize management API: [5] Error establishing socket connection.
[Mar 26 21:36:07.204] {0x2ac3a052d880} NOTE: logging to 0x818260
[Mar 26 21:36:07.204] {0x2ac3a052d880} ERROR: wrote crash log to /var/jenkins/workspace/autest-github/4304/install/var/log/trafficserver/crash-2018-03-26-213607.log
traffic_server: received signal 11 (Segmentation fault)
```
Generally this indicates a failure to put test support for a new configuration var. I don't see that in this PR so it may be an artifact of an earlier PR that slipped through.

**Comment by SolidWallOfCode** [comment]
Woah. I ran the `-R ` regression on master and no failure. I tried the same with this PR and I get the segfault in the same place. It's in `__memmove_avx_unaligned_erms` but the rest of the stack is smashed. It's 59 more frames of `in ??
 ()`.

**Comment by bryancall** [comment]
Actually there might be a bug in TextView::take_prefix_at() in creating an invalid string_view under the hood.

**Comment by bryancall** [comment]
The problem is that the ATS's string_view does a size check in remove_prefix() and gcc doesn't.  I made a PR for that fix #3356.  I am temporary including the fix here to make sure CI will pass with it.

**Pull request merged**